rabbitmq/tests/test_openmetrics.py

```diff
@@ -687,3 +689,42 @@ def test_aggregated_and_detailed_endpoints(query, metrics, aggregator, dd_run_ch
     for m in ['rabbitmq.build_info', 'rabbitmq.identity_info']:
         aggregator.assert_metric_has_tag(m, f'endpoint:http://localhost:15692/metrics/{detailed_ep}', count=1)
         aggregator.assert_metric_has_tag(m, 'endpoint:http://localhost:15692/metrics', count=1)
+
+
+@pytest.mark.parametrize(
+    'plugin_config, err',
+    [
+        pytest.param({}, r'prometheus_plugin\.url field is required\.', id="No URL supplied."),
+        pytest.param(
+            {'url': 'localhost'},
+            r'prometheus_plugin\.url field must be an HTTP or HTTPS URL\.',
+            id="URL supplied without HTTP(S) protocol.",
+        ),
+        pytest.param(
+            {'url': "http://localhost", "unaggregated_endpoint": "detailed?"},
+            r"prometheus_plugin\.unaggregated_endpoint must be either 'per-object' "
+            + r"or 'detailed' or 'detailed\?<QUERY>'\.",
+            id="Invalid nonempty unaggregated_endpoint value.",
+        ),
+        pytest.param(
+            {'url': "http://localhost", "unaggregated_endpoint": ""},
+            r"prometheus_plugin\.unaggregated_endpoint must be either 'per-object' "
+            + r"or 'detailed' or 'detailed\?<QUERY>'\.",
+            id="Empty unaggregated_endpoint value is invalid.",
+        ),
+        pytest.param(
+            {'url': "http://localhost", "unaggregated_endpoint": []},
+            r"prometheus_plugin\.unaggregated_endpoint must be a string\.",
+            id="Unaggregated_endpoint value must be a string.",
+        ),
+        pytest.param(
+            {'url': "http://localhost", "include_aggregated_endpoint": 1},
+            r"prometheus_plugin\.include_aggregated_endpoint must be a boolean\.",
+            id="Aggregated_endpoint must be a boolean.",
```

Comment:
When you remove the type and required checks from the custom validator, you may need to update some of the expected results. They'll produce slightly different errors from what you have currently